src/argus_htmx/utils.py

```diff
@@ -0,0 +1,87 @@
+# Should be in argus proper and merged with api bulk actions
+#
+# THIS IS LESS DRY-ABLE THAN YOU (or SonarCloud) THINK(S)
+#
+# Uses composition, see bulk_change_incidents()
+
+from typing import List, Dict, Any
+
+from django.utils import timezone
+
+from argus.incident.models import Incident
+from argus.util.signals import bulk_changed
+
+
+def send_changed_incidents(incidents):
+    bulk_changed.send(sender=Incident, instances=incidents)
+
+
+def get_qs_for_incident_ids(incident_ids: List[int], qs=None):
+    # setup
+    if qs is None:
+        qs = Incident.objects.all()
+    qs = qs.filter(pk__in=incident_ids)
+    found_ids = set(qs.values_list("id", flat=True))
+
+    # wash ids
+    missing_ids = set(incident_ids) - found_ids
+    return qs, missing_ids
+
+
+def bulk_ack_queryset(actor, qs, data: Dict[str, Any]):
+    timestamp = data["timestamp"]
+    description = data.get("description", "")
+    expiration = data.get("expiration", None)
+    acks = qs.create_acks(actor, timestamp, description, expiration)
+    incidents = []
+    for ack in acks:
+        incidents.append(ack.event.incident)
+    return incidents
```

Comment:
We get the user's group from `actor` but how do we pass in the group the ack is for? It can only go in data.

This needs more thinking so let's not add the complexity now.